src/wallet/wallet.cpp

```diff
@@ -3361,6 +3361,18 @@ void CWallet::AutoCombineDust()
         return;
     }
 
+    if (nAutoCombineBlockFrequency != 0) {
+        // If the block height hasn't exceeded our frequency; or is not a multiple of our frequency.
+        if ((nAutoCombineBlockFrequency > chainActive.Tip()->nHeight) ||
+            (chainActive.Tip()->nHeight % nAutoCombineBlockFrequency)) {
```

Comment:
Instead of call `chainActive.Tip()` again, use the variable `tip` that it's at the beginning of the method.